### PR TITLE
add exec command for crictl

### PR DESCRIPTION
**Pull request by heartlock**
Signed-off-by: heartlock <21521209@zju.edu.cn>

ref: https://github.com/kubernetes-incubator/cri-tools/issues/37
@feiskyer @Random-Liu

### CODE REVIEW

File: cmd/crictl/exec.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+	r, err := client.Exec(context.Background(), &pb.ExecRequest{
+		ContainerId: opts.id,
+		Cmd:         opts.cmd,
+		Tty:         opts.tty,
+		Stdin:       opts.stdin,
+	})
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Url: %v\n", r.Url)
```
Comment (xlgao-zju) [inline cmd/crictl/exec.go]: @feiskyer Should we read the content of the url?

**Comment by heartlock** [comment]
Something weird happened @feiskyer :
image

**Comment by heartlock** [comment]
go vet failed when use newest golang version.

**Comment by feiskyer** [comment]
Opened #47 to fix travis issues.

File: cmd/crictl/exec.go (hunk 1 above)
Comment (feiskyer) [inline cmd/crictl/exec.go]: Yes, I think so. Could you check https://github.com/kubernetes/kubernetes/blob/master/pkg/client/unversioned/remotecommand/remotecommand.go to visit the URL?

**Comment by xlgao-zju** [comment]
@feiskyer Cool!

File: cmd/crictl/exec.go (hunk 1 above)
Comment (xlgao-zju) [inline cmd/crictl/exec.go]: @feiskyer  Will do.

File: cmd/crictl/exec.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
+package main
```
Comment (xlgao-zju) [inline cmd/crictl/exec.go]: @feiskyer Do we need the k8s license?

Comment (feiskyer) [inline cmd/crictl/exec.go]: yes, of course

File: cmd/crictl/exec.go (hunk 1 above)
Comment (heartlock) [inline cmd/crictl/exec.go]: If we need to check url returned available here?

File: cmd/crictl/exec.go (hunk 2 above)
Comment (feiskyer) [inline cmd/crictl/exec.go]: Hmm, there is something wrong of verifiers, will fix later.

Comment (xlgao-zju) [inline cmd/crictl/exec.go]: @heartlock Should add k8s license here.

Comment (heartlock) [inline cmd/crictl/exec.go]: license added @xlgao-zju

File: cmd/crictl/exec.go (hunk 1 above)
Comment (feiskyer) [inline cmd/crictl/exec.go]: @heartlock Yes, please

**Comment by feiskyer** [comment]
@heartlock Any updates on this?

**Comment by heartlock** [comment]
Not yet! I will do it as soon as possible @feiskyer

**Comment by feiskyer** [comment]
@heartlock Please rebase and add streaming support, refer https://github.com/feiskyer/cri-tools/commit/803c07516e44daa5a3b735395be225d86e5a8a3d#diff-d1f71bbe5606eb567619242dad1d9797R142

**Comment by xlgao-zju** [comment]
@heartlock after #58 will import the deps which you need.

**Comment by heartlock** [comment]
ok, I will update this pr after #58 merge

**Comment by heartlock** [comment]
updated @feiskyer

File: cmd/crictl/exec.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+	r, err := client.Exec(context.Background(), &pb.ExecRequest{
+		ContainerId: opts.id,
+		Cmd:         opts.cmd,
+		Tty:         opts.tty,
+		Stdin:       opts.stdin,
+	})
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Url: %v\n", r.Url)
```
Comment (feiskyer) [inline cmd/crictl/exec.go]: Only print this when `--debug` is set.

Comment (heartlock) [inline cmd/crictl/exec.go]: @feiskyer updated! 
will handle `--debug` flag in another pr later

**Comment by heartlock** [comment]
@feiskyer  updated!
will handle  flag `--debug` in another pr

**Pull request merged**